spec/unit/util/windows/service_spec.rb

```diff
@@ -90,262 +98,370 @@ def expect_successful_config_query_and_return(query_return)
     end
   end
 
-  describe "#start" do
+  # This shared example contains the unit tests for the send_service_control_signal
+  # helper as used by service actions like #stop. Before including this example in
+  # your tests, be sure to define the following variables in a `let` context:
+  #     * signal        -- The service control signal to send
+  #     * initial_state -- The service's initial state before sending the signal
+  #     * final_state   -- The service's final state after sending the signal
+  #
+  shared_examples "a service action that sends a service control signal" do |action|
+    it "raises a Puppet::Error if ControlService returns false" do
+      expect_successful_status_query_and_return(dwCurrentState: initial_state)
 
-    context "when the service control manager cannot be opened" do
-      let(:scm) { FFI::Pointer::NULL_HANDLE }
-      it "raises a puppet error" do
-        expect{ subject.start(mock_service_name) }.to raise_error(Puppet::Error)
+      subject.stubs(:ControlService).with(service, signal, anything).returns(FFI::WIN32_FALSE)
+
+      expect { subject.send(action, mock_service_name) }.to raise_error do |error|
+        expect(error).to be_a(Puppet::Error)
+
+        expect(error.message).to match(subject::SERVICE_CONTROL_SIGNALS[signal].to_s)
```

Comment:
TBH I don't like matching on any part of the error message. I understand where you're trying to go with this (i.e. trying to remove false positives where something else errors and the test passes) but I don't think it's worth making the test brittle.

I've seen a couple new tests already break, especially with i18n enabled windows machines (which we test in both unit and acceptance).

I believe it's acceptable risk to just unit test that an error is raised.

Reply:
Yeah that's fine. I'll go ahead and update the PR to remove the error message matching